File: restaurant-client/src/main/java/com/jdc/restaurant/client/utils/DateSerializer.java
package com.jdc.restaurant.client.utils;

import java.io.IOException;
import java.util.Date;

import com.fasterxml.jackson.core.JsonGenerator;
import com.fasterxml.jackson.databind.JsonSerializer;
import com.fasterxml.jackson.databind.SerializerProvider;

public class DateSerializer extends JsonSerializer<Date> {

	@Override
	public void serialize(Date value, JsonGenerator gen, SerializerProvider serializers) throws IOException {
		if(null != value) {
			gen.writeString(DateFormatForJson.df.format(value));
		}
	}

}

File: restaurant-mobile/app/src/main/java/com/jdc/restaurant/api/utils/DateSerializer.kt
package com.jdc.restaurant.api.utils

import com.fasterxml.jackson.core.JsonGenerator
import com.fasterxml.jackson.databind.JsonSerializer
import com.fasterxml.jackson.databind.SerializerProvider
import java.util.*

class DateSerializer: JsonSerializer<Date>() {

    override fun serialize(value: Date?, gen: JsonGenerator?, serializers: SerializerProvider?) {
        if(null != value && null != gen) {
            gen.writeString(dateFormat.format(value))
        }
    }
}